src/views/Search.vue

```diff
@@ -97,11 +98,18 @@ export default defineComponent({
           component: Scanner,
         });
         modal.onDidDismiss()
-        .then((result) => {
+        .then(async (result) => {
           //result : value of the scanned barcode/QRcode
-          if(result.role){
+          if(result.role && result.role !== 'backdrop'){
             this.queryString = result.role
-            this.getProducts(process.env.VUE_APP_VIEW_SIZE, 0);
+            await this.getProducts(process.env.VUE_APP_VIEW_SIZE, 0);
+            this.router.push({ path: `/count/${this.products[0].sku}` });
+            // We are using the value in getProducts function above, and hence
+            // are emptying it after usage only.
+            this.queryString = '';
```

Comment:
I think this needs to do differently, we tried to use the existing method which is quite good, but as the flow works differently we should have a new implementation for the scanning and query search.